app_dart/lib/src/model/ci_yaml/ci_yaml.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:github/github.dart';
+
+import '../proto/internal/scheduler.pb.dart' as pb;
+import 'target.dart';
+
+///
+///
+/// This is a wrapper class around the underlying protos.
+///
+/// See //CI_YAML.md for high level documentation.
+class CiYaml {
+  CiYaml({
+    required this.config,
+    required this.slug,
+    required this.branch,
+  });
+
+  /// The underlying protobuf that contains the raw data from .ci.yaml.
+  final pb.SchedulerConfig config;
+
+  /// The [RepositorySlug] that [config] is from.
+  final RepositorySlug slug;
+
+  /// The git branch currently being scheduled against.
+  final String branch;
+
+  /// Gets all [Target] that run on presubmit for this config.
+  List<Target> get presubmitTargets {
+    if (!config.enabledBranches.contains(branch)) {
+      throw Exception('$branch is not enabled for this .ci.yaml.\nAdd it to run tests against this PR.');
+    }
+    final Iterable<Target> presubmitTargets =
+        _targets.where((Target target) => target.value.presubmit && !target.value.bringup);
+
+    return _filterEnabledTargets(presubmitTargets);
+  }
+
+  /// Gets all [Target] that run on postsubmit for this config.
+  List<Target> get postsubmitTargets {
+    final Iterable<Target> postsubmitTargets = _targets.where((Target target) => target.value.postsubmit);
+
+    return _filterEnabledTargets(postsubmitTargets);
+  }
+
+  /// Filters [targets] to those that should be started immediately.
+  ///
+  /// Targets with dependencies are triggered when there dependency pushes a notification that it has finished.
```

Comment:
I was confused by the `dependencies` with `properties->dependencies`. Nice to expand a little bit.

Reply:
Thanks, it is difficult. I added a pointer saying not to confuse it with the properties->dependencies. I was talking with fujino@ a bit about how we could fix forward, by adding a field like flutter_deps or runtime_deps to the protobuf.